src/main/java/org/osiam/storage/query/UserSimpleFilterChain.java

```diff
@@ -44,53 +43,42 @@
     private ExtensionQueryField extensionFilterField;
 
     public UserSimpleFilterChain(CriteriaBuilder criteriaBuilder, ExtensionDao extensionDao,
-                                 ScimExpression scimExpression) {
+                                 FilterExpression filterExpression) {
         this.criteriaBuilder = criteriaBuilder;
         this.extensionDao = extensionDao;
-        this.scimExpression = scimExpression;
+        this.filterExpression = filterExpression;
 
-        String field = scimExpression.getField();
 
-        userFilterField = UserQueryField.fromString(field.toLowerCase(Locale.ENGLISH));
-
-        // It's not a known user field, so try to build a extension filter
-        if (userFilterField == null) {
-            extensionFilterField = getExtensionFilterField(field.toLowerCase(Locale.ENGLISH));
-            if (extensionFilterField == null) {
-                throw new IllegalArgumentException(String.format("Filtering not possible: '%s' not available", field));
-            }
+        if (filterExpression.getField().isExtension()) {
+            extensionFilterField = getExtensionFilterField(filterExpression);
+            userFilterField = null;
```

Comment:
Yes you are right. I'm not very happy with the nullity though. This is definitely a candidate for optionals in the Java8 PR